rb/lib/selenium/webdriver/remote/w3c_capabilities.rb

```diff
@@ -0,0 +1,204 @@
+            :accept_ssl_certs => false,
+            :takes_screenshot => false,
+            :takes_element_screenshot => false,
+            :page_load_strategy => false,
+            :proxy => nil
+        }
+
+        DEFAULTS.each_key do |key|
+          define_method key do
+            @capabilities.fetch(key)
+          end
+
+          define_method "#{key}=" do |value|
+            @capabilities[key] = value
+          end
+        end
+
+        alias_method :version, :browser_version
+
+        #
+        # Convenience methods for the common choices.
+        #
+
+        class << self
+
+          def edge(opts = {})
+            new({
+                    :browser_name => "MicrosoftEdge",
+                    :platform => :windows,
+                }.merge(opts))
+          end
+
+          def firefox(opts = {})
+            new({
+                    :browser_name => "firefox"
+                }.merge(opts))
+          end
+
+          def w3c?(opts={})
+            return true if opts[:browser_name] == :nightly
+            intersect = Remote::W3CCapabilities::DEFAULTS.keys & opts.keys
+            common = [:takes_screenshot, :proxy]
+            !(intersect - common).empty? || Firefox::Binary.version > 43
+          end
+
+          #
+          # @api private
+          #
+
+          def json_create(data)
+            data = data.dup
+
+            caps = new
+            caps.browser_name = data.delete("browserName")
+            caps.browser_version = data.delete("browserVersion")
+            caps.platform_name = data.delete("platformName").downcase.to_sym if data.has_key?('platform')
+            caps.platform_version = data.delete("platformVersion")
+            caps.accept_ssl_certs = data.delete("acceptSslCerts")
+            caps.takes_screenshot = data.delete("takesScreenshot  ")
+            caps.takes_element_screenshot = data.delete("takesElementScreenshot")
+            caps.page_load_strategy = data.delete("pageLoadStrategy")
+            caps.proxy = Proxy.json_create(data['proxy']) if data.has_key?('proxy')
+
+            # any remaining pairs will be added as is, with no conversion
+            caps.merge!(data)
+
+            caps
+          end
+        end
+
+        # @option :browser_name           [String] required browser name
+        # @option :browser_version                [String] required browser version number
+        # @option :platform_name               [Symbol] one of :any, :win, :mac, or :x
+        # @option :platform_version     [String] required platform version number
+        # @option :accept_ssl_certs  [Boolean] does the driver accept SSL Cerfifications?
+        # @option :takes_screenshot       [Boolean] can this driver take screenshots?
+        # @option :takes_element_screenshot       [Boolean] can this driver take element screenshots?
+        # @option :proxy                  [Selenium::WebDriver::Proxy, Hash] proxy configuration
+        #
+        # @api public
```

Comment:
I'm not sure, it is literally copied over from existing code - https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/remote/capabilities.rb#L193
I don't know how yard uses the public/private designations.